### PR TITLE
Fixes race condition when adding additional Windows HDDs

### PR DESCRIPTION
**Pull request by nakomis**

### CODE REVIEW

File: blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java [hunk 1]
```diff
@@ -98,12 +97,36 @@ public void createFilesystem(AttachedBlockDevice attachedDevice, FilesystemOptio
...
+                    "$disk = Receive-Job -Job $job -Wait; " +
+                    "$job = New-Partition -InputObject $disk " + driveLetterParam + " -UseMaximumSize -AsJob; " +
+                    "Wait-Job $job; " +
+                    "$partition = Receive-Job -Job $job -Wait; " +
+                    "$job = Format-Volume -DriveLetter $partition.DriveLetter -FileSystem " + filesystemOptions.getFilesystemType() +" -NewFileSystemLabel \"" + volumeLabelParam + "\" -Confirm:$false -AsJob; " +
+                    "Wait-Job $job; " +
+                    "Receive-Job -Job $job -Wait";
+            builder.append("Initializing disk:\n");
+            builder.append(command);
+            WinRmToolResponse response = ((WinRmMachineLocation)machine).executePsScript(command);
```
Comment (richardcloudsoft) [inline blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java]: (minor) I note that you are building a log string with the execution in the middle, and then issuing the  LOG just once at the end. Recommend you do 2 LOG statements, one before the execution and one after. That way if `executePsScript` exceptions, or something else unusual happens, you have at least half of the log message in the log!

File: blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java [hunk 2]
```diff
@@ -98,12 +97,36 @@ public void createFilesystem(AttachedBlockDevice attachedDevice, FilesystemOptio
...
+                    "do {" +
+                    "  $job = Get-Disk -AsJob; " +
+                    "  Wait-Job $job; " +
+                    "  $disks = Receive-Job -Job $job -Wait; " +
                     // AWS May create an instance store volume, which we need to exclude
                     // See https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/ec2-windows-volumes.html#instance-store-volume-map
-                    "Where {$_.SerialNumber -As [int] -NotIn 78..89 -Or $_.FriendlyName -Ne \"AWS PVDISK\"} | " +
-                    "Initialize-Disk -PartitionStyle MBR -PassThru | New-Partition " + driveLetterParam + " -UseMaximumSize | " +
-                    "Format-Volume -FileSystem " + filesystemOptions.getFilesystemType() +" -NewFileSystemLabel \"" + volumeLabelParam + "\" -Confirm:$false");
+                    "  $rawDisk = $disks | Where PartitionStyle -eq 'RAW' | Where {$_.SerialNumber -As [int] -NotIn 78..89 -Or $_.FriendlyName -Ne \"AWS PVDISK\"} | Select-Object -first 1; " +
```
Comment (richardcloudsoft) [inline blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java]: It looks like the old version of this code could support multiple additional disks, but the new version will only support one disk. Is this a regression - did it use to be possible for the user to add multiple additional disks? Is this a configuration we are supporting?

File: blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java (hunk 1 above)
Comment (nakomis) [reply blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java]: Oops, that was my intention!

File: blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java (hunk 2 above)
Comment (nakomis) [reply blockstore/src/main/java/brooklyn/location/blockstore/AbstractVolumeManager.java]: The old version was in error - it was doing part of the processing on two disks at the same time, whereas this method is called for each additional volume so should only process on at a time. This assumes (hopefully safely) that the volumes returned by the `Get-Disk` scriptlet are in the same order as the volumes in the YAML. Unfortunately there's no way to identify which is which other than the order in which they appear. I've manually run a `Get-Disk | Format-List *`, and there are no properties that can tie them back to the YAML

**Pull request merged**